File: Culinario_DB/EFCore/Tables/Comment.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Culinario_DB.EFCore.Tables;

[Table("Comment")]
public class Comment : ITable
{
    /// <summary>
    /// Уникальный идентификатор комментария
    /// </summary>
    [Key]
    public int Id { get; set; }

    /// <summary>
    /// Текстовое содержимое комментария
    /// </summary>
    [MaxLength(5000)]
    public required string Text { get; set; }

    /// <summary>
    /// Идентификатор рецепта, с которым связан комментарий
    /// </summary>
    public required Recipe Recipe { get; set; }

    /// <summary>
    /// Идентификатор пользователя, создавшего комментарий
    /// </summary>
    public required User User { get; set; }

    /// <summary>
    /// Временная метка создания комментария
    /// </summary>
    public required DateTime TimeCreated { get; set; }

    /// <summary>
    /// Количество лайков, полученных комментарием
    /// </summary>
    public required uint Likes { get; set; }

    /// <summary>
    /// Количество дизлайков, полученных комментарием
    /// </summary>
    public required uint Dislikes { get; set; }

    /// <summary>
    /// Идентификатор комментария, на который является ответом данный комментарий, если таковой имеется
    /// </summary>
    public Comment? ReplyComment { get; set; }
}

File: Culinario/app/src/main/java/com/culinario/mvp/models/Comment.kt
package com.culinario.mvp.models

import androidx.annotation.Size
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Класс Comment представляет комментарий с уникальным идентификатором, текстом комментария, идентификатором рецепта, идентификатором пользователя, временем создания, количеством лайков, дизлайков и идентификатором комментария, на который был дан ответ.
 * @param Id Уникальный идентификатор комментария.
 * @param Text Текст комментария, длина строки не должна превышать 5000 символов.
 * @param Recipe Идентификатор рецепта, к которому относится комментарий.
 * @param User Идентификатор пользователя, оставившего комментарий.
 * @param TimeCreated Время создания комментария.
 * @param Likes Количество лайков комментария.
 * @param Dislikes Количество дизлайков комментария.
 * @param ReplyComment Идентификатор комментария, на который был дан ответ.
 */
@Serializable
data class Comment(
    override val Id: Int,
    @Size(max = 5000) var Text: String,
    val Recipe: Recipe,
    val User: User,
    var TimeCreated: String,
    var Likes: UInt,
    var Dislikes: UInt,
    val ReplyComment: Comment?,
) : IModel